test/TestBComptroller.spec.ts

```diff
@@ -2,45 +2,575 @@ import * as b from "../types/index";
+      const cZRX_addr = compoundUtil.getContracts("cZRX");
+
+      const avatar1 = await bProtocol.registry.newAvatar.call({ from: a.user1 });
+
+      await bProtocol.registry.newAvatar({ from: a.user1 });
+      expect(await bProtocol.registry.avatarOf(a.user1)).to.be.equal(avatar1);
+
+      expect(await comptroller.checkMembership(avatar1, cZRX_addr)).to.be.equal(false);
+
+      const errArr = await bComptroller.enterMarkets.call([cZRX_addr, cZRX_addr], {
+        from: a.user1,
+      });
+      await Promise.all(errArr.map((e) => expect(e).to.be.bignumber.equal(ZERO)));
+      await bComptroller.enterMarkets([cZRX_addr, cZRX_addr], { from: a.user1 });
+
+      expect(await comptroller.checkMembership(avatar1, cZRX_addr)).to.be.equal(true);
+    });
+
+    it("should fail when BToken not exists for a given market", async () => {
+      const cBAT_addr = compoundUtil.getContracts("cBAT");
+      await expectRevert(
+        bComptroller.enterMarkets([cBAT_addr], { from: a.user1 }),
+        "BComptroller: BToken-not-exist-for-cToken",
+      );
+    });
+  });
+
+  describe("BComptroller.exitMarket()", async () => {
+    beforeEach(async () => {
+      const cZRX_addr = compoundUtil.getContracts("cZRX");
+      const cETH_addr = compoundUtil.getContracts("cETH");
+
+      await bComptroller.newBToken(cZRX_addr);
+      await bComptroller.newBToken(cETH_addr);
+
+      await bComptroller.enterMarkets([cZRX_addr, cETH_addr], { from: a.user1 });
+    });
+
+    it("should fail when user not have avatar", async () => {
+      const cZRX_addr = compoundUtil.getContracts("cZRX");
+
+      await expectRevert.unspecified(bComptroller.exitMarket(cZRX_addr, { from: a.user2 }));
+    });
+
+    it("user can exit from a market he is in", async () => {
+      const cZRX_addr = compoundUtil.getContracts("cZRX");
+      const cETH_addr = compoundUtil.getContracts("cETH");
+
+      const avatar1 = await bProtocol.registry.avatarOf(a.user1);
+      expect(await comptroller.checkMembership(avatar1, cZRX_addr)).to.be.equal(true);
+      expect(await comptroller.checkMembership(avatar1, cETH_addr)).to.be.equal(true);
+
+      const err = await bComptroller.exitMarket.call(cZRX_addr, { from: a.user1 });
+      expect(err).to.be.bignumber.equal(ZERO);
+      await bComptroller.exitMarket(cZRX_addr, { from: a.user1 });
+
+      expect(await comptroller.checkMembership(avatar1, cZRX_addr)).to.be.equal(false);
+      expect(await comptroller.checkMembership(avatar1, cETH_addr)).to.be.equal(true);
+    });
+  });
+
+  describe("BComptroller.getAccountLiquidity()", async () => {
+    beforeEach(async () => {
+      const cZRX_addr = compoundUtil.getContracts("cZRX");
+      const cETH_addr = compoundUtil.getContracts("cETH");
+
+      await bComptroller.newBToken(cZRX_addr);
+      await bComptroller.newBToken(cETH_addr);
+
+      await bComptroller.enterMarkets([cZRX_addr, cETH_addr], { from: a.user1 });
+    });
+
+    it("should get account liquidity of a user");
```

Comment:
why this is empty?

Reply:
These are TODOs, I will come back to these tests once we have other basic tests ready to generate some liquidity for an account.

Those tests will be shown as pending in the test results. This will remind us to complete these pending tests.